src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5900,17 +5900,15 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
             }
 
             greaterThanToken = this.EatToken();
-            return result;
-        }
 
-        private bool DetermineIfDefinitelyTypeArgumentList(bool isDefinitelyTypeArgumentList)
-        {
-            if (!isDefinitelyTypeArgumentList)
+            // If we have `X<Y>)` then this would definitely be a type argument list.
+            isDefinitelyTypeArgumentList = isDefinitelyTypeArgumentList || this.CurrentToken.Kind is SyntaxKind.CloseParenToken;
```

Comment:
Could you help me understand in which scenarios we are depending on this line versus line 5874 for correct behavior?

Reply:
so 5874 related to *incomplete* nested generics.  so if you had `(X<Y<Z>)` (note the lack of the last `>`) . 

This line applies when you have `(X<Y>)` or `(X<Y<Z>>)`

Reply:
thanks!